### PR TITLE
Runs integration test daily

### PR DESCRIPTION
**Pull request by meghfossa**
# Overview

This PR, 
- Fixes broken integration tests. 
- Runs CI integration test on week

From our discussion, we wanted to start running this daily at the start, instead of every push (if I'm remembering correctly). 

## Acceptance criteria

- CI integration tests are run on a daily cadence
- CI integration test pass

## Testing plan

You can manually run this workflow, by
1. Going to Actions
2. Selecting 'Integration Test' and dispatching workflow on this branch. 

Here is the CI run: https://github.com/fossas/fossa-cli/actions/runs/2204419218

## Risks

N/A

## References

N/A

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,16 @@ build:
 test:
 	cabal test unit-tests --test-show-details=streaming --test-option=--format=checks --test-option=--times --test-option=--color
 
+# Runs a integeration test.
+# To run a set of integeration spec matching specific value, use ARGS
+# 	Example: make integeration-test-for ARGS="fd"
+integeration-test:
```
Comment (scruffystuffs) [inline Makefile:13]: Fix the spelling, add more description to example, wording.
```suggestion
# Runs an integration test.
# To run a set of integration tests matching a specific value, use ARGS
# For example, to only run tests whose name matches the wildcard '*fd*':
# 	make integration-test ARGS="fd"
integration-test:
```

File: integration-test/Analysis/ElixirSpec.hs [hunk 2]
```diff
@@ -15,7 +15,7 @@ elixirEnv :: FixtureEnvironment
 elixirEnv = NixEnv ["elixir"]
 
 mixBuildProjectCmd :: Command
-mixBuildProjectCmd = Command "mix" ["deps.get", "&&", "mix deps.compile"] Never
+mixBuildProjectCmd = Command "mix" ["local.hex --force --if-missing", "&&", "mix local.rebar --force", "&&", "mix deps.get", "&&", "mix deps.compile"] Never
```
Comment (scruffystuffs) [inline integration-test/Analysis/ElixirSpec.hs:18]: Let's multi-line this list and leave a comment explaining the options.

**Review by scruffystuffs — APPROVED** [review]
Tiny fix needed before merging.  Can we also check that this doesn't conflict too much with the GHC 9 PR, or just block this until after?  I won't be able to update that PR tomorrow, so I'd prefer to avoid conflicts.

**Comment by meghfossa** [comment]
> Tiny fix needed before merging. Can we also check that this doesn't conflict too much with the GHC 9 PR, or just block this until after? I won't be able to update that PR tomorrow, so I'd prefer to avoid conflicts.

Let's wait until GHC 9 gets merged, I can rebase and merge it after.

**Pull request merged**